### PR TITLE
HSEARCH-2597 Do not rely on automatic detection of JPA annotations fr…

### PR DESCRIPTION
**Pull request by Sanne**
…om the classpath

Resolve some inconsistencies around `@Id` handling, and polish a couple of tricky tests.

See JIRA for the full rationale.

https://hibernate.atlassian.net/browse/HSEARCH-2597

### CODE REVIEW

File: engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java [hunk 1]
```diff
@@ -152,4 +154,14 @@ public BatchBackend getBatchBackend() {
 		return batchBackend;
 	}
 
+	/**
+	 * The default for the tests in hibernate-search-engine is to not rely on JPA
+	 * specific annotations. These are typically used though when integrating
+	 * with Hibernate, so some tests might want to explicitly enable this.
+	 */
+	public SearchFactoryHolder enableJPAAnnotationsProcessing(boolean enableJPAAnnotationsProcessing) {
```
Comment (yrodiere) [inline engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java]: Not related to this PR, but it seems we're introducing two different naming schemes in this class for option setters:

 * `withMultitenancyEnabled`, `withIdProvidedImplicit`, ...
 * `multipleInstances`, `enableJPAAnnotationsProcessing`, ...

No need to change this PR, but just so I know, do you have a preference?

File: engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java [hunk 2]
```diff
@@ -86,6 +87,7 @@ private SearchIntegrator createSearchFactory() {
 		}
 		cfg.setIdProvidedImplicit( idProvidedImplicit );
 		cfg.setMultitenancyEnabled( multitenancyEnabled );
+		cfg.setEnableJPAAnnotationsPRocessing( enableJPAAnnotationsProcessing );
```
Comment (Sanne) [inline engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java]: thanks

File: engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java (hunk 1 above)
Comment (Sanne) [inline engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java]: I find the new name more suited for this specific case, as it has an imperative effect on a "toggle".

You're right it's inconsistent though. I think the only (weak) justification I can think of is that the multi-tenancy and provided-id-strategies need to be plugged in additional strategies to work.

**Comment by Sanne** [comment]
fixed the camel-casing and merged

**Pull request closed without merging**